sdk/eventhub/azure-eventhub/tests/conftest.py

```diff
@@ -122,41 +124,41 @@ def get_credential():
             "Environment variable AZURE_TEST_USE_PWSH_AUTH set to 'true'. Using AzurePowerShellCredential."
         )
         from azure.identity import AzurePowerShellCredential
-        return AzurePowerShellCredential()
+        return AzurePowerShellCredential
     # User-based authentication through Azure CLI, if requested
     elif use_cli.lower() == "true":
         log.info("Environment variable AZURE_TEST_USE_CLI_AUTH set to 'true'. Using AzureCliCredential.")
         from azure.identity import AzureCliCredential
-        return AzureCliCredential()
-    return EnvironmentCredential()
+        return AzureCliCredential
+    return DefaultAzureCredential
 
+@pytest.fixture(scope="session")
```

Comment:
I think you should probably use the [`get_credential`](https://github.com/Azure/azure-sdk-for-python/blob/main/tools/azure-sdk-tools/devtools_testutils/azure_recorded_testcase.py#L193) from our devtools:

`from devtools_testutils import get_credential`

I believe federated auth requires the use of `AzurePipelinesCredential` which the devtools `get_credential` will use in the pipeline. This might be why your test pipelines are failing.